generators/opa5/templates/Journey.js

```diff
@@ -0,0 +1,32 @@
+sap.ui.define([
```

Comment:
If I opt-in for TypeScript and also opa5 I'm afraid I'll end up with js tests in my ts repo because the templates are currently js only 🙈  Same is true for qunit tests

Reply:
Correct, I haven't added ts templates yet. Would you be willing to help with those?

Reply:
Sure. That won't be an issue. But I think we need to answer some questions first:
- Fiori Elements and the OData version are optional
  - with FEv4 you can use `sap.fe.test` (including the `sap.fe.test.JourneyRunner` to run them)
  - with FEv2 you can use `Opa5.createPageObjects` enhanced by the fioriElementsTestLibrary (but afaik `sap.suite.ui.generic.template.integration.testLibrary` can't be served via npmjs; this only works when proxied)
  - without FE you use `Opa5.createPageObjects`
- Is the preview middleware optional?
  - with the preview middleware you have a sandbox launchpad (for tests as well as local execution). So you can use `iStartMyApp` instead of `iStartMyUIComponent` in the OPA5 tests

So we might end up with quite some different templates (but that's true for JS as well)

- Let's assume FEv4 for the next one. Basically I'm aware of two different setups:

  1. having a file in `./page` for each page of the FE floorplan (e.g. List Report) that contains the configuration for the respective page for the Journey Runner (e.g. `new sap.fe.test.ListReport({...})`) and potentially some custom actions/assertions.
The Journey then imports the pages as needed and hands them over to the JourneyRunner. So the JourneyRunner instance is created in the Journey (one Journey one runner instance).
  1. having an `AllJourneys.(j|t)s` that contains all the pages configuration in one single place that also starts the JourneyRunner. So there is exactly one instance of the JourneyRunner running all Journeys.

Not sure which one is to be preferred. I'll get in touch with FE on that one.